### PR TITLE
Sqlite - fix default connect and levenshtein

### LINKED ISSUE #1250: SQLite linker default connection option doesn't work
### What happens?

See [this linker setup code](https://github.com/moj-analytical-services/splink/blob/aba0f5f2ba40958ee642f69df836c1e53cacf9c1/splink/sqlite/sqlite_linker.py#L86-L96) - if we use a `SQLiteLinker` without providing a connection we have the default arg of `:memory:`. However this does not get turned into a connection, just a string, so the `__init__` gives us `AttributeError: 'str' object has no attribute 'row_factory'`.

We should handle the argument so that we instantiate a connection if we are only given a string.

### To Reproduce

```py
import pandas as pd

from splink.sqlite.sqlite_linker import SQLiteLinker

df = pd.DataFrame({"id": [1, 2, 3], "name": ["a", "b", "c"]})
linker = SQLiteLinker(df, {"link_type": "dedupe_only"})
```

### OS:

MacOS

### Splink version:

3.8.1

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by ADBond**
### Type of PR

- [x] BUG
- [x] FEAT
- [ ] MAINT
- [ ] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?
Closes #1250.

Also allows registering of `levenshtein` by default in `SQLiteLinker` - see #1316.

I also notice `rapidfuzz` (which we use here for Levenshtein) also [implements all of the other string metrics we use](https://maxbachmann.github.io/RapidFuzz/Usage/distance/index.html). Realise they are obviously a relatively slow option compared to other backends but could also add these in the same way if it's something that might be useful?

### PR Checklist

- [NA] Added documentation for changes
- [NA] Added feature to example notebooks at tutorials in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
- [x] Added tests (if appropriate)
- [x] Made changes based off the latest version of Splink
- [x] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

**Comment by RossKen** [comment]
I think it would be good to add the other fuzzy matching functions by default (and caveat somewhere that they may not be the most performant)

**Comment by ADBond** [comment]
Re: further fuzzy string matching: have implemented udfs, comparisons and comparisons.

Still to do:
- [ ] update the relevant docs

**Comment by RossKen** [comment]
Hey @ADBond, given the additional fuzzy match functions have been added, we should probably set up the ctl library for sqlite as well. I'm not sure if sqlite has a regex extract functionality, but `name_comparison` and `forename_surname_comparison` should work now given the functionality you have added.

**Comment by ADBond** [comment]
> Hey @ADBond, given the additional fuzzy match functions have been added, we should probably set up the ctl library for sqlite as well. I'm not sure if sqlite has a regex extract functionality, but `name_comparison` and `forename_surname_comparison` should work now given the functionality you have added.

I don't believe it natively has regex functions, but it should be possible to add a udf to capture that behaviour. Might make sense to save that for a separate 'regex' PR that does similar for postgres + converts relevant tests maybe?

**Comment by RossKen** [comment]
> > Hey @ADBond, given the additional fuzzy match functions have been added, we should probably set up the ctl library for sqlite as well. I'm not sure if sqlite has a regex extract functionality, but `name_comparison` and `forename_surname_comparison` should work now given the functionality you have added.
> 
> I don't believe it natively has regex functions, but it should be possible to add a udf to capture that behaviour. Might make sense to save that for a separate 'regex' PR that does similar for postgres + converts relevant tests maybe?

Yea I think that is best left for a separate PR as well 😊

File: splink/sqlite/sqlite_helpers/sqlite_comparison_imports.py [hunk 1]
```diff
@@ -3,15 +3,24 @@
     DistanceFunctionLevelBase,
```
[the remark this answers is not in the document]
Comment (ADBond) [reply splink/sqlite/sqlite_helpers/sqlite_comparison_imports.py:3]: I copied the extra levels from duckdb - looks like in there + in Spark the `damerau_levenshtein_level` is accidentally just a `levenshtein_level` with a different name. Will fix but will also add some tests which have eyes on that.

[the remark this answers is not in the document]
Comment (RossKen) [reply splink/sqlite/sqlite_helpers/sqlite_comparison_imports.py:3]: That is not ideal - thank you for spotting/fixing  👍

**Comment by ADBond** [comment]
> The only thing thing that I had to do was install `rapidfuzz` and it worked fine. Is this the behaviour we want as standard? I know we want to keep dependencies to a minimum, but if we are going to expect people to install rapidfuzz we probably want an explicit error message and/or it documented somewhere that it needs to be installed (plus the caveats on potential performance implications).

This is partly why I included an option for turning the udfs off - in case a user is unable/unwilling to install this extra library. Will add some user-facing info on this

File: splink/sqlite/sqlite_helpers/sqlite_comparison_imports.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ADBond) [reply splink/sqlite/sqlite_helpers/sqlite_comparison_imports.py:4]: this is updated and fixed for the other backends also

**Review by RossKen — APPROVED** [review]
Great job! Good to go from my end 👍

**Pull request merged**